corehq/apps/app_manager/static/app_manager/js/app_manager.js

```diff
@@ -376,6 +376,7 @@ hqDefine('app_manager/js/app_manager', function () {
                             var requiresCaseDetails = hqImport('app_manager/js/details/screen_config').state.requiresCaseDetails;
                             requiresCaseDetails(data['case_list-show']);
                         }
+                        $form.trigger('saved-app-manager-form');
```

Comment:
Interesting approach. Custom events should be used sparingly, but this seems fine to me.